modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/persistence/RocksDbKeyValueStorage.java

```diff
@@ -783,6 +783,35 @@ public void removeAll(List<byte[]> keys, KeyValueUpdateContext context) {
         }
     }
 
+    @Override
+    public void removeAll(byte[] prefix, KeyValueUpdateContext context) {
+        rwLock.writeLock().lock();
+
+        try (WriteBatch batch = new WriteBatch(); Cursor<Entry> entries = range(prefix, nextKey(prefix))) {
+            long curRev = rev + 1;
+
+            List<byte[]> keys = new ArrayList<>();
```

Comment:
I think that using a collection here can be memory intensive and I think that we can delete immediately by cursor.

Reply:
Only for RocksDB, yes? For SimpleInMemoryKeyValueStorage it seems impossible to avoid accumulating keys to a collection